OptimoveSDK/Sources/Classes/Optimobile/InApp/InAppPresenter.swift

```diff
@@ -88,29 +103,44 @@ class InAppPresenter : NSObject, WKScriptMessageHandler, WKNavigationDelegate{
             }
         }
 
+        let notShowingCurrentTickle = self.currentMessage != nil
+            && self.currentMessage!.id != (self.messageQueue[0] as! InAppMessage).id
+            && (self.messageQueue[0] as! InAppMessage).id == self.pendingTickleIds[0] as! Int64
+
+        let queueNotEmptyAndNotShowingAnything = self.currentMessage == nil && self.messageQueue.count > 0
+
+        let shouldShowSomething = notShowingCurrentTickle || queueNotEmptyAndNotShowingAnything
+
         messageQueueLock.signal()
-      
-        DispatchQueue.main.async {
-            self.initViews()
-            
-            if (self.currentMessage != nil
-                && self.currentMessage!.id != (self.messageQueue[0] as! InAppMessage).id
-                && (self.messageQueue[0] as! InAppMessage).id == self.pendingTickleIds[0] as! Int64) {
```

Comment:
This logic is concerned with the following edge-case:

- the app is showing an in-app message
- the user taps a tickle for a different in-app
- the user should see the tapped tickle immediately
- the SDK may still be loading the message referenced by the tickle (hence 'pending')